### PR TITLE
Fixed conversion from matrix3 to axis angle

### PR DESCRIPTION
**Pull request by omichel**
The edge cases where &theta; = &pi; where not correctly computed (error in the formula) in #2113.
Thanks to @Simon-Steinmann, I investigated this method and found the error based on his edge case test:
```
-1  0  0
 0  0  1
 0  1  0

=> { (0, 0.7071068, 0.7071068) 3.1415927 }
```
Other cases can be quickly verified with https://www.andre-gaschler.com/rotationconverter/

### CODE REVIEW

File: src/webots/maths/WbRotation.cpp [hunk 1]
```diff
@@ -46,41 +48,34 @@ void WbRotation::fromQuaternion(const WbQuaternion &q) {
...
-    mZ = M(1, 0) - M(0, 1);
-    normalizeAxis();
-    mAngle = theta;
-  } else {
-    // If `theta == pi`
+    return;
+  } else if (fabs(theta - M_PI) < WbPrecision::DOUBLE_EQUALITY_TOLERANCE) {  // If `theta == pi`
     if (M(0, 0) > M(1, 1) && M(0, 0) > M(2, 2)) {
-      mX = sqrt(M(0, 0) - M(1, 1) - M(2, 2) + 0.5);
+      mX = sqrt(M(0, 0) - M(1, 1) - M(2, 2) + 1) / 2;
```
Comment (DavidMansolino) [inline src/webots/maths/WbRotation.cpp:60]: Well spotted!

**Review by DavidMansolino — APPROVED** [review]
Thank you.
